src/common/io/tls/session.c

```diff
@@ -60,7 +80,7 @@ tlsSessionClose(TlsSession *this, bool shutdown)
     if (this->session != NULL)
     {
         // Shutdown on request
-        if (shutdown)
+        if (this->shutdownOnClose)
             SSL_shutdown(this->session);
 
         // Free the socket session
```

Comment:
My comment is for lines 91 and 92 - here the calling order is:
```
memContextCallbackClear(this->memContext);
tlsSessionFreeResource(this);
```
but in storageReadPosixClose it is called in the opposite order:
```
storageReadPosixFreeResource(this);
memContextCallbackClear(this->memContext);
```
Are they both correct? If so, why does order not matter?

Reply:
Hmm, I'd say it is correct here and wrong in storageReadPosixClose(). Clearing the callback is so the callback won't be called if there is an error in storageReadPosixFreeResource(), since this is the same function will be called on error.

Corrected in storageReadPosixClose() here 0b349c0f2f54d1.